### PR TITLE
Add CMS Price Check default CDS Service for invocation on medication prescribe hook

### PR DESCRIPTION
**Pull request by zplata**
Currently, the default CDS Service `cms-price-check` has code hosted within the Sandbox repository. This service is invoked on the `medication-prescribe` hook, and specifically, when a medication is authored on the Rx View of the Sandbox. The service presents a price of a drug chosen as a response, or a cheaper option if one is available through a suggestion. 

With the intent to extract the CDS Services code from the Sandbox into this repository, we should move the `cms-price-check` default service here. One specific change added here from the service hosted on the Sandbox repository is that instead of rounding the price of each medication to the nearest dollar, we should round to the nearest cent and display medication prices to the cent for more accuracy (i.e. Try `Tylenol` on the Sandbox and take the suggestion, noting the before and after pricees). Additionally, fixing minor bugs here for missing cases in some special-case drugs to make sure if a price is available, it will be returned.

The service parses the context block of the CDS service request based on the new expectation for `medication-prescribe` in the PR [here](https://github.com/cds-hooks/docs/pull/109).

### CODE REVIEW

File: services/cms-price-check.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+function getValidContextResources(request) {
+  const resources = [];
+  const { context } = request.body;
+  if (context && context.patientId && context.medications) {
+    const medResources = context.medications;
+    medResources.forEach((resource) => {
+      const validResource = getValidResource(resource, context);
+      if (validResource) {
+        resources.push(resource);
```
Comment (shriniketsarkar) [inline services/cms-price-check.js]: Should we be pushing the `validResource` instead of the resource here ?

File: services/cms-price-check.js [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+          resource: suggestionResource,
+        },
+      ],
+    }];
+  }
+  return card;
+}
+
+function getSuggestionCard(prices, genericCode, resource) {
+  if (prices.generic && prices.brand) {
```
Comment (shriniketsarkar) [inline services/cms-price-check.js]: Is `prices` guaranteed to be available ? 
If not checking for its existence would be nice to have : 
`if (prices && prices.generic && prices.brand) {`

Comment (zplata) [reply services/cms-price-check.js]: Yes, it's checked [here](https://github.com/cds-hooks/sandbox-cds-services/pull/7/files#diff-ad2cbff78178a04861ae72b427e24768R129) and we return null before the method is called

Comment (shriniketsarkar) [reply services/cms-price-check.js]: 👍

File: services/cms-price-check.js (hunk 1 above)
Comment (zplata) [reply services/cms-price-check.js]: No, we should push the entire resource so it is processed later when making the CDS Hooks card. I do see how the variable name could make it misleading, so I re-named it for it's real purpose [here](https://github.com/cds-hooks/sandbox-cds-services/pull/7/commits/546ebd819522858ca31dd4912ada441cc7b466ba). Really, `getValidResource` is a chain of methods to check if the `resource` variable is valid enough to process for a possible card. Thanks for catching that!

**Pull request merged**